### ISSUE TITLE
UUID type

### ISSUE THREAD

**Issue by DaanRademaker** [mitchelllisle/sparkdantic#232]
Great looking library. Would love to use it in my projects but I am missing UUID type. Would you be open to support UUID type (that simply maps to a spark StringType() )

**Comment by DaanRademaker** [comment]
Hi @mitchelllisle, 
Yes support for own type mappings would definitely be a great solution. 
Are you open to contributions? Or is this something you would like to implement yourself?

**Comment by mitchelllisle** [comment]
More than happy to accept contributions. 

Would be good to discuss what you're thinking first so we can align on approach but all for people contributing

**Comment by mitchelllisle** [comment]
@DaanRademaker 

I've gone and implemented this in the latest release #250 

```python
class MyModel(SparkModel):
    id: UUID4 = Field(spark_type=StringType)
    t: str
```

This produces the following schema:

```
StructType([
    StructField('id', StringType(), False),
    StructField('t', IntegerType(), False)
])
```

**Comment by DaanRademaker** [comment]
Awesome, thanks!
